siliconcompiler/apps/sc_remote.py

```diff
@@ -0,0 +1,113 @@
+import os
+import sys
+import time
+
+from siliconcompiler import Chip
+from siliconcompiler.remote.client import (cancel_job, check_progress, delete_job,
+                                           remote_ping, remote_run_loop)
+from siliconcompiler.utils import default_credentials_file
+
+
+def main():
+    progname = "sc-remote"
+    description = """
+    -----------------------------------------------------------
+    SC app that provides an entry point to common remote / server
+    interactions. Can be used to:
+    * Check software versions on the server (no flags)
+    * Check an ongoing job's progress (-jobid)
+    * Cancel an ongoing job (-jobid + -cancel)
+    * Re-attach SC client to an ongoing job (-jobid + -attach)
+    -----------------------------------------------------------
+    """
+
+    # Argument Parser
+    parser = argparse.ArgumentParser(prog=progname,
+                                     formatter_class=argparse.RawDescriptionHelpFormatter,
+                                     description=description)
+    parser.add_argument('-jobid', required=False)
+    parser.add_argument('-cfg', required=False)
+    parser.add_argument('-reconnect', action='store_true', required=False)
+    parser.add_argument('-cancel', action='store_true', required=False)
+    parser.add_argument('-delete', action='store_true', required=False)
+    parser.add_argument('-credentials', nargs='?', default=default_credentials_file())
+
+    args = parser.parse_args()
+
+    # Sanity checks.
+    if (args.reconnect and (args.cancel or args.delete)):
+        print('Error: -reconnect is mutually exclusive to -cancel and -delete')
+        return 1
+    elif (args.cancel and (args.reconnect or args.delete)):
+        print('Error: -cancel is mutually exclusive to -reconnect and -delete')
+        return 1
+    elif (args.delete and (args.reconnect or args.cancel)):
+        print('Error: -delete is mutually exclusive to -reconnect and -cancel')
+        return 1
+    elif ((args.reconnect or args.cancel or args.delete) and not args.jobid):
+        print('Error: -jobid is required for -reconnect, -cancel, and -delete')
+        return 1
+    elif (args.reconnect and not args.cfg):
+        print("Error: -cfg is required for -reconnect. Recommended value is "
+              "the post-import manifest in the job's original build directory.")
+        return 1
+
+    # Read in credentials from file, if specified and available.
+    # Otherwise, use the default server address.
+    if os.path.isfile(args.credentials):
+        with open(args.credentials, 'r') as cfgf:
+            try:
+                remote_cfg = json.loads(cfgf.read())
+            except json.JSONDecodeError:
+                print('Error reading remote configuration file: invalid JSON')
+                return 1
+    else:
+        # TODO: I think this default is stored somewhere - client.py? _metadata.py?
+        remote_cfg = {'address': 'https://server.siliconcompiler.com',
```

Comment:
it's in _metadata, please use the client to retrieve this information or rely on it to add the `remote_cfg` and `jobhash` to the chip. This will simpliyf removing the `status` dict later